### PR TITLE
Fix for WF agent not recognising walls and initial wall following position

### PR DESCRIPTION
**Pull request by laurajogi**
WF agent now detects walls properly again and should act reliably. Also each agent detects its own initial position where it started wall following.

### CODE REVIEW

**Comment by laurajogi** [comment]
The WF agent works as it's supposed to (finally) and starts following walls again if it finds an unexplored one. Will create different variants of the agent for experiments in a new branch now.

File: src/main/java/app/model/Map.java [hunk 1]
```diff
@@ -80,7 +81,8 @@ public Map(Settings settings)
         {
             Vector srt = randPosition(intruderSpawn);
             Vector dir = randDirection();
-            Agent intruder = new DirectionFollowAgent(srt, dir, 10, Type.INTRUDER, targetDirection(srt));
+            //Agent intruder = new DirectionFollowAgent(srt, dir, 10, Type.INTRUDER, targetDirection(srt));
```
Comment (S010MON) [inline src/main/java/app/model/Map.java:84]: Is this code being used?  If not it's probably best to remove to keep it clean

**Review by S010MON — CHANGES_REQUESTED** [review]
This seems to fix the issue, but I am getting weird freezing after running the simulation for more than 20 seconds.  There might be a memory leak somewhere that is causing this?  I don't know if anyone else is having the same issue, but this doesn't occur on the `master` branch.

**Comment by S010MON** [comment]
https://user-images.githubusercontent.com/10490509/168468036-d54f4dd4-0d92-4e5d-8559-bd7a193a0ae8.mp4

This is a video of the issues I'm getting

**Comment by S010MON** [comment]
Screenshot from 2022-05-16 16-03-40

So the slowdowns seems to be eating a load of CPU, but the memory isn't changed much.  SO not a memory leak, but likely you're doing a really expensive operation that should be checked out later on.

I've approved the PR

**Review by S010MON — APPROVED** [review]

**Review by S010MON — DISMISSED** [review]

**Review by trav-d13 — APPROVED** [review]
Looks good. Runs well my side

**Review by MatthewFoster02 — APPROVED** [review]

**Pull request merged**